sematic/resolvers/local_resolver.py

```diff
@@ -444,7 +442,7 @@ def _populate_graph(
             # Updating the input artifact
             artifact_id = None
 
-            maybe_resolved_value = future.resolved_kwargs.get(name, value)
+            maybe_resolved_value = future.props.get_resolved_kwargs().get(name, value)
             if not isinstance(maybe_resolved_value, AbstractFuture):
                 artifact = self._make_artifact(
```

Comment:
Food for thought that the `ExternalResource` inputs will need special treatment.

The lifecycle of the resource will be managed by the Resolver, for the Future in which it was created. The resource will be used in a descendant Future, according to our planned usage pattern.

During normal execution, the lifecycle is managed in concurrence with the updating of the resource state. But for reran and cached runs, the relationship will be severed. When rerunning a future that uses an external resource, its ancestor which created the resource will need to be reran as well.

Reply:
This is a great call-out! Though the ancestor didn't actually activate the resource, it just made it such that the resolver knew about the configuration of the resource and the fact that the future needed the resource. So we will need some way during the "cloning" step of rerun-from-here to clone this dependency on the resource. I think that should be pretty doable, but we will need to look out for it 👀 .